### PR TITLE
import into: redact task meta after import & cancel on kill

### PR DESCRIPTION
**Pull request by D3Hunter**
### What problem does this PR solve?

Issue Number: ref #42930 

Problem Summary:

### What is changed and how it works?
- unify and fix log, avoid exposing meta info in log
- redact meta after task end(success or err)
- cancel job on kill event
- change default `thread` param to `0.5*cpuCount`
- add some missed ut

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [x] Manual test (add detailed scripts or steps below)
  - add some wait before start sorting, start import, kill it in another terminal, the job should be cancelled, check task status too, should be reverted
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: disttask/importinto/dispatcher.go [hunk 1]
```diff
@@ -590,12 +594,22 @@ func (h *flowHandle) failJob(ctx context.Context, handle dispatcher.TaskHandle,
...
+	redactSensitiveInfo(gTask, taskMeta)
 	return handle.WithNewSession(func(se sessionctx.Context) error {
 		exec := se.(sqlexec.SQLExecutor)
 		return importer.FailJob(ctx, exec, taskMeta.JobID, errorMsg)
 	})
 }
 
+func redactSensitiveInfo(gTask *proto.Task, taskMeta *TaskMeta) {
+	taskMeta.Stmt = ""
+	taskMeta.Plan.Path = ast.RedactURL(taskMeta.Plan.Path)
```
Comment (Ehco1996) [inline disttask/importinto/dispatcher.go:606]: After looking at the redactURL implementation, which currently only supports s3, do we need to consider the gcs case?

https://github.com/pingcap/tidb/blob/8633c8ce9921bf17a92b582a6b5d299d95213ac6/parser/ast/misc.go#L3438

Comment (D3Hunter) [reply disttask/importinto/dispatcher.go:606]: no

Comment (D3Hunter) [reply disttask/importinto/dispatcher.go:606]: it's for url like `s3://`, not http

**Review by GMHDBJD — APPROVED** [review]
LGTM

**Review by lance6716 — APPROVED** [review]

**Pull request merged**